app_flutter/lib/logic/task_grid_filter.dart

```diff
@@ -0,0 +1,232 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:collection';
+
+import 'package:flutter/material.dart';
+
+import 'package:cocoon_service/protos.dart' show CommitStatus;
+
+import '../logic/qualified_task.dart';
+import '../widgets/filter_property_sheet.dart';
+
+/// A filter object for controlling which entries are visible in the Build dashboard grid
+/// of tasks. This filter object can filter on a number of properties of the tasks including
+/// the name of the task, information on the commit that generated the tasks, and platform
+/// stage type toggles.
+class TaskGridFilter extends FilterPropertySource {
+  TaskGridFilter();
+
+  factory TaskGridFilter.fromMap(Map<String, String> valueMap) => TaskGridFilter()..applyMap(valueMap);
+
+  /// True iff all of the properties of the filter are set to their default values.
+  bool get isDefault =>
+      _allProperties.values.toList().every((ValueFilterProperty<dynamic> element) => element.isDefault);
+
+  /// Sets all properties of this filter to their default values.
+  void reset() {
+    for (final ValueFilterProperty<dynamic> property in _allProperties.values) {
+      property.reset();
+    }
+  }
+
+  /// Modifies this filter based on the values in the map. If the map is null, no changes are
+  /// made. All keys in the [valueMap] must match one of the field names of the filter properties.
+  void applyMap(Map<String, String> valueMap) {
+    if (valueMap != null) {
+      for (final MapEntry<String, String> mapEntry in valueMap.entries) {
```

Comment:
`valueMap?.entries`?

Reply:
You can't iterate on null. Is there an `in?` operator?

Reply:
Ah I didn't know. I've been spending too much in python. I was naive in thinking that it would just work.